src/test/java/ChatServerTest.java

```diff
@@ -0,0 +1,89 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.concurrent.Executors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ChatServerTest {
+    private ByteArrayOutputStream recordedOutput;
+    private PrintStream output;
+    private ServerSocket serverSocket;
+    private Socket socket;
+
+    @Before public void setUp() throws IOException {
+        recordedOutput = new ByteArrayOutputStream();
+        output = new PrintStream(recordedOutput);
+        serverSocket = new ServerSocket(4444);
+        socket = new Socket("localhost", 4444);
+    }
+
+    @Test
+    public void makesAConnection() throws IOException {
+        UserIO console = createConsole("quit\n");
```

Comment:
These tests feel like integration tests, where you are starting up the client and server, and testing the output of the server. There is value in having those tests, and it has resulted in you having a fairly simple design for your app as you're mainly using the Java IO libraries. However I would like to see more unit tests that don't require you to boot up the main parts of your app.